### PR TITLE
Load buses into ports table

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Enable to show buses in Ports table of pin assignment feature.

 ### Which part of the code base require a change
 
 - [x] Library: pinassignment
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### Screenshots
image

### CODE REVIEW

File: tests/unittest/PinAssignment/PortsLoader_test.cpp [hunk 1]
```diff
@@ -68,3 +70,24 @@ TEST(PortsLoader, LoadAllData) {
     EXPECT_EQ(port3.range, "Msb: 2, lsb: 0") << port3.range.toStdString();
   }
 }
+
+TEST(PortsLoader, LoadBus) {
+  PortsModel model;
+  PortsLoader loader{&model};
+  auto [res, error] = loader.load(":/PinAssignment/ports_test.json");
```
Comment (ravic-rs) [inline tests/unittest/PinAssignment/PortsLoader_test.cpp:77]: Can you link me to this json ? I didn't find it under /src/PinAssignment.

Comment (KochynVolodymyr) [reply tests/unittest/PinAssignment/PortsLoader_test.cpp:77]: it is not under src, https://github.com/os-fpga/FOEDAG/blob/main/tests/unittest/PinAssignment/ports_test.json

File: src/PinAssignment/PortsView.cpp [hunk 2]
```diff
@@ -45,26 +44,14 @@ PortsView::PortsView(PinsBaseModel *model, QWidget *parent)
...
-
-      setItemWidget(item, PackagePinCol, combo);
-      model->portsModel()->insert(item->text(PortName),
-                                  indexFromItem(item, PortName));
+      if (p.isBus) {
+        auto item = new QTreeWidgetItem;
+        item->setText(PortName, p.name);
+        topLevel->addChild(item);
+        for (const auto &subPort : p.ports) insertTableItem(item, subPort);
+      } else {
```
Comment (ravic-rs) [inline src/PinAssignment/PortsView.cpp:52]: What's this if/else for ?

Comment (KochynVolodymyr) [reply src/PinAssignment/PortsView.cpp:52]: if bus else not bus

File: src/PinAssignment/PortsModel.cpp [hunk 3]
```diff
@@ -26,7 +26,7 @@ PortsModel::PortsModel(QObject *parent)
     : QObject(parent), m_model(new QStringListModel(this)) {}
 
 QStringList PortsModel::headerList() const {
-  return {"Name", "Dir", "Package Pin", "Type", "Range"};
+  return {"Name", "Dir", "Package Pin", "Type"};
```
Comment (ravic-rs) [inline src/PinAssignment/PortsModel.cpp:29]: Don't we need range anymore ?

Comment (KochynVolodymyr) [reply src/PinAssignment/PortsModel.cpp:29]: I don't think so. This is an information for build buses. And it is reflected in the table.

Comment (ravic-rs) [reply src/PinAssignment/PortsModel.cpp:29]: Seems like we need "Range" after speaking to Tao. Let's connect tomorrow to discuss offline.

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**